distributed/worker.py

```diff
@@ -298,7 +298,11 @@ def _close(self, report=True, timeout=10):
             v.stop()
         self.rpc.close()
         self.status = 'closed'
-        self.stop()
+        with ignoring(gen.TimeoutError):
+            yield gen.with_timeout(timedelta(seconds=1),
+                    [close(stream) for stream in self._listen_streams])
+        for stream in self._listen_streams:
+            stream.close()
```

Comment:
In `distributed/tests/test_stress.py::test_stress_scatter_death` we close workers during computation.  We were finding that connections to killed workers were becoming non-responsive but still remained open.  The living worker trying to gather data from the dead worker would wait for a very long time on a living but ineffective stream.